Sources/MongoSwift/MongoCollection+Read.swift

```diff
@@ -66,6 +67,31 @@ extension MongoCollection {
         return Int(count)
     }
 
+    /**
+     * Counts the number of documents in this collection matching the provided filter.
+     *
+     * - Parameters:
+     *   - filter: a `Document`, the filter that documents must match in order to be counted
+     *   - options: Optional `CountOptions` to use when executing the command
+     *
+     * - Returns: The count of the documents that matched the filter
+     */
+    public func countDocuments(_ filter: Document = [:], options: CountOptions? = nil) throws -> Int {
+        throw MongoError.commandError(message: "Unimplemented command")
```

Comment:
Should this have a todo linked to a JIRA ticket? Likewise for `estimatedDocumentCount` below.